packages/codegen.go/src/models.ts

```diff
@@ -443,87 +449,110 @@ function generateJSONUnmarshallerBody(modelType: go.Model | go.PolymorphicModel,
     addlPropsText += `${tab}\t\t\t${receiver}.AdditionalProperties[key] = ${assignment}\n`;
     addlPropsText += `${tab}\t\t}\n`;
     addlPropsText += `${tab}\t\tdelete(rawMsg, key)\n`;
+    needsErrCheck = true;
     return addlPropsText;
   };
-  let unmarshalBody = '';
-  unmarshalBody = '\tfor key, val := range rawMsg {\n';
-  unmarshalBody += '\t\tvar err error\n';
-  unmarshalBody += '\t\tswitch key {\n';
-  let addlProps: go.Map | undefined;
-  for (const field of values(modelType.fields)) {
-    if (field.type.kind === 'map' && field.annotations.isAdditionalProperties) {
-      addlProps = field.type;
-      continue;
-    }
-    unmarshalBody += `\t\tcase "${field.serializedName}":\n`;
-    if (hasDiscriminatorInterface(field.type)) {
-      unmarshalBody += generateDiscriminatorUnmarshaller(field, receiver);
-    } else if (field.type.kind === 'time') {
-      unmarshalBody += `\t\t\t\terr = unpopulate${capitalize(field.type.format)}(val, "${field.name}", &${receiver}.${field.name})\n`;
-      modelDef.SerDe.needsJSONUnpopulate = true;
-    } else if (field.type.kind === 'slice' && field.type.elementType.kind === 'time') {
-      imports.add('time');
-      let elementPtr = '*';
-      if (field.type.elementTypeByValue) {
-        elementPtr = '';
+
+  const emitSwitchCase = function(): string {
```

Comment:
This change just stuffs the original code into this helper function and sets `needsErrCheck` as required. The rest of the code is untouched.